libs/wire-subsystems/src/Wire/UserSubsystem.hs

```diff
@@ -160,6 +160,11 @@ data UserSubsystem m a where
   InternalUpdateSearchIndex :: UserId -> UserSubsystem m ()
   InternalFindTeamInvitation :: Maybe EmailKey -> InvitationCode -> UserSubsystem m StoredInvitation
   GetUserExportData :: UserId -> UserSubsystem m (Maybe TeamExportUser)
+  -- | Throws errors in the interpreter.
+  RemoveEmail :: Local UserId -> UserSubsystem m ()
```

Comment:
i would make this a helper function that uses the UserSubsystem effect constraint to call RemoveEmailEither.  i can't think of a reason to keep the flexibility here and pay with the extra complexity inside the effect.  plus it's not going to hard to change it back in the future, should the need arise.

also it is more modular: the error that `RemoveEmail` throws won't depend on the UserSubsystem interpreter, but on whichever interpreter is plugged in to throw.  (not sure i'm making sense to myself, but i think the point i'm trying to make is valid.  maybe let's call on monday?)

Reply:
I believe the changes to it are in line with what you had in mind, but please confirm.